File: analytical-web/src/app/models/quiz/Quiz.ts
import { QueryDocumentSnapshot } from '@angular/fire/firestore';
import { Questions } from '../Questions';

export interface Quiz {
  id: string;
  title: string;
  desc: string;
  cover_photo: string;
  subject: 'MATH' | 'ENGLISH';
  schoolLevel: 'SHS' | 'JHS';
  category: Questions['type'];
  levels: number;
  visible: boolean;
  createdAt: Date;
}

// export enum Categories {
//   REBUS_PUZZLE = 'REBUS_PUZZLE',
//   RIDDLES = 'RIDDLES',
//   WORD_PUZZLE = 'WORD PUZZLE',
//   MATH_LOGIC_PUZZLE = 'MATH LOGIC PUZZLE',
// }

// export enum Subjects {
//   MATH = 'MATH',
//   ENGLISH = 'ENGLISH',
// }
export const quizConverter = {
  toFirestore: (data: Quiz) => data,
  fromFirestore: (snap: QueryDocumentSnapshot) => {
    const quiz = snap.data() as Quiz;
    quiz.createdAt = (quiz.createdAt as any).toDate();
    return quiz;
  },
};

File: Analytical/app/src/main/java/com/ketchupzzz/analytical/models/quiz/Quiz.kt
package com.ketchupzzz.analytical.models.quiz

import android.os.Parcelable
import com.ketchupzzz.analytical.models.Category
import com.ketchupzzz.analytical.models.Subject
import kotlinx.parcelize.Parcelize
import java.util.Date

@Parcelize
data class Quiz(
    val id : String ? = null,
    val title : String? = null,
    val desc : String ? = null,
    val cover_photo : String ? = null,
    val subject : Subject? = null,
    val category: Category? = null,
    val levels : Int ? = null,
    val visible: Boolean ? = null,
    val createdAt : Date? = null,
) : Parcelable


@Parcelize
data class Levels(
    val id : String = "",
    val name : String = "",
    val levelNumber : Int = 0,
    val points :  Int = 0,
    val questions : Int = 0,
    val timer : Int = 0,
) : Parcelable